docs/source/class_source.rst

```diff
@@ -0,0 +1,335 @@
+When a version is not provided, ``DefaultClassAvgSource`` defaults to
+the latest version available.
+
+
+Classifiers
+***********
+
+Classifiers take an image ``Source`` and attempts to classify into
+``classes`` that identify images with similar viewing angles up to
+reflection.  All ``Class2D`` instances are expected to implement a
+``classify`` method which returns ``(classes, reflections,
+distances)``.  The three returned variables are expected to be 2D
+Numpy arrays in a neighbor network format having shape ``(n_classes,
+n_nbors)``.  So to retrieve the input source indices for the first
+class's neighbors, we would want ``classes[0,:]``.  The first class
+index should be the base image used for classification.  So we would
+expect ``classes[0,0]`` to be ``input_src.images[0]``.
+
+No further class selection or order occurs during classification.
+Those methods are broken out into other components.
+
+Currently ASPIRE has a single classification algorithm known as
+``RIRClass2D``.  This algorithm uses multiple applications of PCA in
+conjunction with Bispectrum analysis to identify nearest neighbors in
+a rotationally invariant feature space.
+
+.. mermaid::
+
+   classDiagram
+      class Class2D{
+	  +classify()
+      }
+    Class2D <|-- RIRClass2D
+
+Class Selectors
+***************
+
+Class Selectors consume the output of ``Class2D`` and attempt to order
+and/or filter classes down to a selection.  Selecting the "best"
+classes in CryoEM problems is still an area of active research.  Some
+common methods are provided, along with an extensible base interface.
+
+Generally, Class Selection comes in two flavors depending on what
+information is required to perform the selection.
+
+Local Class Selectors
+---------------------
+
+For "Local" class selection, we will attempt to use only the
+information returned from ``Class2D``.  In the case of ``RIRClass2D``
+this would primarily be a network of ``distances`` as measured in the
+compressed feature space.
+
+This approach has two main advantages.  First, we already have this
+information computed as part of classification.  Second, it allows us
+to register and stack a relatively small subset of the "best" classes.
+Because registration and alignment are computationally expensive this
+can reduce pipeline run times by an order of magnitude.
+
+.. mermaid::
+
+   classDiagram
+      class ClassSelector{
+	 +select()
+	 }
+       ClassSelector <|-- TopClassSelector
+       ClassSelector <|-- RandomClassSelector
+       ClassSelector <|-- NeighborVarianceClassSelector
+       ClassSelector <|-- DistanceClassSelector
+       ClassSelector o-- GreedyClassRepulsion
+
+Global Class Selectors
+----------------------
+
+Global Class Selection techniques first compute the entire collection
+registered and aligned class averages, then compute some quality
+measure on all classes.
+
+These are most similar to the historical MATLAB approaches, sometimes
```

Comment:
Right, but I think legacy is the only reason to use this... I have confirmed it seems inferior in practice (with real, non-simulated data) to other methods created in this PR.

I do want to keep it, if only to test claims of past papers that specifically state this is their technique.

Reworded to avoid MATLAB.  Not sure if out-of-core was just a Y-ism (isn't how I use that term) so removed as well.

Reply:
Ok sounds good.